algorithms.py

```diff
@@ -1457,6 +1457,20 @@ def hudson_recombination_event(self, label, return_heads=False):
             ret = x, alpha
         return ret
 
+    def generate_gc_tract_length(self):
+        # generate tract length
+        if self.discrete_genome:
+            tl = np.random.geometric(1 / self.tract_length)
+        else:
```

Comment:
I'm being dense here @fbaumdicker, but why not use ``tl = np.random.exponential(self.tract_length)``?

```
>>> np.mean(np.random.exponential(10, size=10**7))
10.002678522591678
>>> np.mean(np.random.exponential(1, size=10**7))
0.9999924193218064
>>> np.mean(np.random.exponential(0.1, size=10**7))
0.10002349776505161
```

This takes away all the singularities, and gives a nice simple way to explain what it's doing: When we have ``discrete_genome=False`` the gene conversion tract lengths are drawn from an exponential distribution with rate equal to the ``tract_length`` parameter. When ``discrete_genome`` is True, the values are drawn from a geometric distribution with parameter 1 / tract_length. 

This way the expected length of gene conversion tracts is always equal to ``tract_length``, right?

Reply:
Yes, that would be the case.

I am mainly thinking in discrete coordinates, so I am really unsure what the standard user would expect if he is simulating a continuous genome. My reason to transform the mean of the exponential is mainly based on the idea that I would transform the coordinates into discrete coordinates after the simulation such that the left and right gene conversion tract ends would need to be rounded (to the previous and next larger integer). I wanted the mean of the resulting rounded tracts to be the same as the supplied parameter.
But maybe that is not what other users would expect? How should users interpret continuous gene conversion tracts?
Does a tract from 4.234 to 6.421 correspond to a tract from 4 to 6 or from 4 to 7?
Is a tract from 4.234 to 4.83 a gene conversion event at all?
Or should we think of continuous coordinates as the limit for large genome sizes anyway, such that even 4.234 to 4.235 covers a bunch of bases?

if we use the original tract_length and if users transform their coordinates after the simulation, the results will differ compared to the discrete case. I wanted to avoid that. In particular, for tract_length = 1 this can lead to very different results.
The geometric distribution with mean 1 has zero variance, as the tract length always equals 1 in this case. For the exponential distribution tract lengths above 1 are possible. 
However, if users will not transform their coordinates then all of this is not an issue.

I would recommend making the decision, based on what you think people are using continuous coordinates for and whether you want the discrete and continuous case to be interchangeable.

Reply:
We should view continuous coordinates as just that, and assume that users will leave the coordinates as-is afterwards for analysis. I would imagine the majority of people working in continuous coordinates would work in the unit interval.

Put it this way - recombination already acts on continuous coordinates, and we can have millions of recombinations in a 10bp interval. It would be very confusing and surprising if gene conversion behaved differently - there's nothing inherently discrete above GC above and beyond recombination. If recombination can be assumed to occur at fractional genome locations, then gc breakpoints should be assumed to occur at fractional genome locations, and there's certainly no reason why gc tract lengths would have be to of integer lengths (particuarly and especially since the breakpoints would start at continuous points in the first place).

Do you see where I'm coming from? The default will be to work in discrete space as you prefer, but there's no reason this can't also work in continuous space in an analogous way to recombination.

Reply:
Ok, I am convinced. Will change the mean of the continuous tract lengths back to tract_length. I will also remove the requirement for the tract_length to be >= 1 in the continuous case. Larger than 0 should suffice in this case. Especially if people use the unit interval as their coordinates.